docs/control_structures.rst

```diff
@@ -296,9 +296,11 @@ Subroutines
 .. note::
     Subroutines are only available in TEAL version 4 or higher.
 
-A subroutine is section of code that can be called multiple times from within a program. Subroutines
-are PyTeal's equivalent to functions. Subroutines can accept any number of arguments, and these
-arguments must be PyTeal expressions. Additionally, a subroutine may return a single value, or no value.
+A subroutine is section of code that can be called multiple times from within a program. Subroutines are PyTeal's equivalent to functions.  Subroutine constraints include:
+
+* Subroutines accept any number of arguments.
+* Subroutine argument types can be `Expr` (PyTeal expression) or `ScratchVar`.  PyTeal applies pass-by-value semantics to `Expr` and pass-by-reference to `ScratchVar`.
```

Comment:
```suggestion
* Subroutine argument types can be `Expr` (PyTeal expression) or `ScratchVar` (though `ScratchVar` subtypes such as `DynamicScratchVar`'s are currently disallowed).  PyTeal applies pass-by-value semantics to `Expr` and pass-by-reference to `ScratchVar`.
```
... to be 100% transparent and up to date. But this is also a moving target.

Reply:
@tzaffi Agree on the subclass restriction callout.  I made a minor tweak to the suggested verbiage in https://github.com/algorand/pyteal/pull/226/commits/2afc82a33839fbf9ad10b57aee0bbe1d9b07c3e3.